File: The_Name_Game/Perl/name.pl
sub printVerse {
    $name = shift;
    $sb = lc($name);
    $x = ucfirst $sb;
    $x0 = substr $x, 0, 1;

    if ($x0 eq 'A' or $x0 eq 'E' or $x0 eq 'I' or $x0 eq 'O' or $x0 eq 'U') {
        $y = lc($x);
    } else {
        $y = substr $x, 1;
    }

    $b = 'b' . $y;
    $f = 'f' . $y;
    $m = 'm' . $y;

    if ($x0 eq 'B') {
        $b = $y;
    } elsif ($x0 eq 'F') {
        $f = $y;
    } elsif ($x0 eq 'M') {
        $m = $y;
    }

    print "$x, $x, bo-$b\n";
    print "Banana-fana fo-$f\n";
    print "Fee-fi-mo-$m\n";
    print "$x!\n\n";
}

@nameList = ( "Gary", "Earl", "Billy", "Felix", "Mary", "Steve" );

foreach $name (@nameList) {
    printVerse($name);
}

File: The_Name_Game/Ruby/name.rb
def printVerse(name)
    x = name.capitalize
    y = case x[0]
        when 'A','E','I','O','U' then name.downcase
        else x[1..-1]
        end
    b = 'b' + y
    f = 'f' + y
    m = 'm' + y
    case x[0]
    when 'B' then b = y
    when 'F' then f = y
    when 'M' then m = y
    end

    print x, ", ", x, ", bo-", b, "\n"
    print "Banana-fana fo-", f, "\n"
    print "Fee-fi-mo-", m, "\n"
    print x, "!\n\n"
end

for name in ["Gary","Earl","Billy","Felix","Mary","Steve"]
    printVerse(name)
end
